extensions/amp-iframe/0.1/test/test-amp-iframe.js

```diff
@@ -76,6 +79,9 @@ describes.realWin('amp-iframe', {
         }
       });
       setTrackingIframeTimeoutForTesting(20);
+      setTrackingIframeCountForTesting(0);
+      sandbox.stub(error, 'reportError');
```

Comment:
@rsimha I encountered the issue in #15658 as well. 
However the test is intentionally trying to test the misconfiguration. I used `allowConsoleError`, but that doesn't work because `user().assert()` call `#reportError()`, and that will rethrow the error again async.
This is my temporarily solution to fix this, stub `#reportError()`, I think a more general solution would be to stub `#reportError()` for all tests. WDYT

Reply:
You can use `expectAsyncConsoleError(<message>)` for async errors. Did you try that?

It's explained in the failure message.

Reply:
I hesitate to stub out `reportError` since several tests actually check if it fires. However, we could make it so that the error is thrown synchronously during tests.

Reply:
And finally, if a mocha test does something asynchronous, it is responsible for calling the `done()` callback to indicate that the test has completed. Unfortunately, a lot of our tests fail to do so.

Reply:
> You can use expectAsyncConsoleError(<message>) for async errors. Did you try that?

I tried that. It doesn't work because I didn't call `done()` and the test ends earlier.

I think it will works if I call the `#done()` callback. But the problem is that I don't know where to call the `done()` callback if I can't stub the `#reportError()`. Can I do `yield expectAsyncConsoleError(message)`. 

The test logic is super simple, check if a misconfiguration results in a function not being called eventually. I don't think it makes much sense to have these unit tests figure out the logic behind `#assert()` and call `#done()` callback to not break tests.

Reply:
Talked offline. Will leave the stub there until we figure out a nice general solution.

Reply:
Meanwhile, #16005 should fix the `layoutCallback` issues we were discussing here.